core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -489,7 +489,6 @@ sealed class Vec[T <: Data] private[chisel3] (gen: => T, val length: Int) extend
             s"are less than zero or greater or equal to than Vec length"
         )
       }
-      cloneSupertype(elementInitializers.map(_._2), s"Vec.Lit(...)")
```

Comment:
This was being used as a check, but it is what was blocking size-0 from working in the `Vec(0, ...).Lit` use case. Rather, that check is only needed when using the `Vec.Lit(...)` API (since no element type is given) so I basically moved this check there but also went ahead and used it to figure out the resulting `Vec` type.